plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergConfig.java

```diff
@@ -26,6 +26,7 @@
 {
     private IcebergFileFormat fileFormat = ORC;
     private HiveCompressionCodec compressionCodec = GZIP;
+    private boolean useFileSizeFromMetadata = true;
```

Comment:
Have you confirmed this is fixed in the current Spark writer? If this is only a problem for old files, or people using old writers, I'd rather make this `false` by default. Let people knowingly opt-in to the performance hit if they have a problem.

I feel like if we make this `true` now, we won't know when it is safe to turn off, especially since the Iceberg user base will grow over time.

Reply:
@electrum 

> If this is only a problem for old files, or people using old writers, I'd rather make this false by default.

Not sure if I understand this correctly. Shouldn't it be the other way since `useFileSizeFromMetadata = true` means trusting the writer's value? May be the config name is confusing, should we change it to `useFileSizeFromManifest`?

The fix for one of the issues reported so far https://github.com/apache/iceberg/issues/1980 is yet to be released. So I'd say that the problem exists in the latest release of spark writer.

 > I feel like if we make this true now, we won't know when it is safe to turn off, especially since the Iceberg user base will grow over time.

That was my thought around trusting the stored value by default. But since the current writer can have issues, may be we can set `useFileSizeFromMetadata = false` by default. Once 0.11 is released and updated in spark, we can change the default to `true` and drop the config eventually.

Reply:
I'd suggest setting the default to `true`, meaning trust the writer. Also, 0.11 should be released this month.

Reply:
I misread this and got it backwards. I agree with the default of true (trust the writer).